File: mock-esp/src/model/Request.cpp
#include "Request.h"

Request::Request(String method, String path, std::map<String, String> headers, String body) {
  this->method = method;
  this->path = path;
  this->headers = headers;
  this->body = body;
}

String Request::getMethod() {
  return this->method;
}

String Request::getPath() {
  return this->path;
}

std::map<String, String> Request::getHeaders() {
  return this->headers;
}

String Request::getBody() {
  return this->body;
}

File: mock-server/src/main/java/model/Request.java
package model;

import java.util.Collections;
import java.util.Map;
import java.util.TreeMap;

import static java.util.Objects.requireNonNull;

public class Request {
    private String method;
    private String path;
    private Map<String, String> headers;
    private String body;

    public Request(String method, String path, Map<String, String> headers, String body) {
        this.method = requireNonNull(method, "Parameter method is required.");
        if(method.isEmpty()) throw new IllegalArgumentException("Parameter method cannot be empty.");
        this.path = requireNonNull(path, "Parameter path is required.");
        this.headers = (headers != null) ? headers : new TreeMap<>();
        this.body = (body != null) ? body : "";
        this.body = this.body.replace("\\r\\n", String.valueOf('\n'));
        this.body = this.body.replace("\\n", String.valueOf('\n'));
    }

    public String getMethod() {
        return method;
    }

    public String getPath() {
        return path;
    }

    public Map<String, String> getHeaders() {
        return Collections.unmodifiableMap(headers);
    }

    public String getBody() {
        return body;
    }
}
